renderite/generator/RenderiteGenerator/Generator/TypeGenerators/PackableStructGenerator.cs

```diff
@@ -0,0 +1,193 @@
+        {
+            // if (this._ilVerbose)
+                // w.Any($"{instruction} ({instruction.OpCode.FlowControl})");
+
+            FlowControl flow = instruction.OpCode.FlowControl;
+            if (flow != FlowControl.Next && flow != FlowControl.Call && flow != FlowControl.Return)
+            {
+                w.Fixme($"Unknown {flow} instruction");
+                w.Comment(instruction.ToString());
+            }
+
+            if (instruction.OpCode.Code is Code.Ldfld or Code.Ldflda)
+            {
+                string name = ((FieldReference)instruction.Operand).Name;
+                names.Enqueue(name);
+                // if (this._ilVerbose)
+                    // w.Any($"{instruction.OpCode.Code} {name}");
+            }
+
+            if (instruction.OpCode.Code is Code.Call && instruction.Operand is MethodReference callRef)
+            {
+                switch (callRef.Name)
+                {
+                    // Write single value or packed boolean
+                    case "Write" when callRef.Parameters.Count == 1:
+                    case "WriteObject":
+                    {
+                        string name = names.DequeueLast();
+                        w.Any($"try ipc.write(@TypeOf(self.{name}), self.{name});");
+                        written = true;
+                        break;
+                    }
+                    case "Write" when callRef.Parameters.All(p => p.ParameterType.Name == "Boolean"):
+                    {
+                        List<string> paramsList = names.Select(n => "self." + n).ToList();
+                        while (paramsList.Count < 8)
+                        {
+                            paramsList.Add("false");
+                        }
+                        w.Any($"try ipc.write{paramsList.Count}PackedBools({string.Join(", ", paramsList)});");
+                        names.Clear();
+                        written = true;
+                        break;
+                    }
+                    // Read single value or packed boolean
+                    case "Read" when callRef.Parameters.Count == 1:
+                    case "ReadObject":
+                    {
+                        string name = names.DequeueLast();
+                        w.Any($"self.{name} = try ipc.read(@TypeOf(self.{name}));");
+                        written = true;
+                        break;
+                    }
+                    case "Read" when callRef.Parameters.All(p => p.ParameterType.Name == "Boolean&"):
+                    {
+                        List<string> paramsList = names.Select(n => "self." + n).ToList();
+                        // TODO: ceil to nearest multiple of 8
+                        // FE doesn't pack more than 1 byte at a time so its not particularly important
+                        while (paramsList.Count < 8)
+                        {
+                            paramsList.Add("_");
+                        }
```

Comment:
You should add an assertion here to throw an error if it *is* over 8, just in case.